lib/libc/gen/dlfcn.c

```diff
@@ -202,9 +203,9 @@ dl_init_phdr_info(void)
 		case AT_BASE:
 			phdr_info.dlpi_addr =
 #ifdef __CHERI_PURE_CAPABILITY__
-			    /* XXXAR: currently needs load_cap for libunwind */
-			    (uintptr_t)cheri_andperm(auxp->a_un.a_ptr,
-			        CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP);
+			    (uintptr_t)(Elf_Addr)auxp->a_un.a_ptr;
+			assert(phdr_info.dlpi_addr == 0 &&
+			    "Should be zero for static binaries");
```

Comment:
Something to bear in mind: this assertion makes it a flag day, so if we do need to do any Release 1 updates in FETT then we'll need to include an updated kernel.

Reply:
I think they should already have been zero just potentially tagged zero?

Reply:
Oh I see